cpp/src/slice2js/Gen.cpp

```diff
@@ -510,47 +489,43 @@ Slice::JsVisitor::writeConstantValue(
     return os.str();
 }
 
-StringList
-Slice::JsVisitor::splitComment(const ContainedPtr& p)
+void
+Slice::JsVisitor::writeDocCommentFor(const ContainedPtr& p, bool includeDeprecated)
```

Comment:
This `includeDeprecated` parameter doesn't sound correct to me.

The generation of deprecated doc-comment tags should be totally independent of whether the Contained is deprecated.

Reply:
I didn't look too closely at this function, but I think this field might still be useful.

To my knowledge, javascript doesn't have a standalone attribute like `[Obsolete]` or `[[deprecated]]`.
The only way to mark something deprecated is at the doc-comment level, so maybe linters and other tools will check for this doc comment tag?

And there are some things that we don't want to mark as deprecated, like the generated service skeletons.

Reply:
So you want to add a special JS-only logic like:  Container is deprecated but there is no @deprecated doc-comment => generate @deprecated JS doc-comment?

Sounds ok, but definitely deserves a comment.

Then:
> And there are some things that we don't want to mark as deprecated, like the generated service skeletons.

Yes, I actually did something similar for C++:
https://github.com/zeroc-ice/ice/blob/e5261ffac22399d93315fa70fa22595c3a40ad24/cpp/src/slice2cpp/Gen.cpp#L370

Reply:
> So you want to add a special JS-only logic like

No, the inverse of that. I want to add exactly what you described for C++!
I was just justifying why we might want that behavior, because I was unaware it already existed :)

I'll co-opt your code into the parser, and then use it for all the different languages.
But since this PR is already pretty big, I'd prefer to leave that for when I refactor all the `deprecated` stuff,
since there's like 3 separate issues for cleaning up how `deprecated` works...

Reply:
Say you define in Slice:
```slice
/// The base interface for all Widgets 
["deprecated"] interface Widget { ... }
```

with a doc-comment but for `@deprecated` doc-comment tag (which is fine at the Slice level). What do you want to generate for the corresponding proxy in JS and TS?

Since JS/TS doesn't provide a deprecated attribute (as far as I can tell), generating an `@deprecated `JS/TS doc-comment seems better than not generating anything.

Reply:
Ok, please create an issue to fix it in a follow-up PR.